source/features/mute-users.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import features from '../libs/features';
+import {appendBefore} from '../libs/dom-utils';
+import optionsStorage, {Options} from '../options-storage';
+import {getUsername} from '../libs/utils';
+
+let currentUser: string;
+
+async function getMutedUsers(): Promise<string[]> {
+	return (await optionsStorage.getAll() as Options).mutedUsers.split(/\s+/);
+}
+
+async function setMutedUsers(mutedUsers: string[]): Promise<void> {
+	return optionsStorage.set({mutedUsers: mutedUsers.join(' ')});
+}
+
+function minimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.remove('d-none');
+	select('.minimized-comment .Details-element', comment)!.removeAttribute('open');
+	select('.unminimized-comment', comment)!.classList.add('d-none');
+}
+
+function unminimizeComment(comment: HTMLElement): void {
+	if (select.exists('.js-targetable-comment[id^="issue-"]', comment)) {
+		return;
+	}
+
+	select('.minimized-comment', comment)!.classList.add('d-none');
+	select('.minimized-comment .Details-element', comment)!.setAttribute('open', 'true');
+	select('.unminimized-comment', comment)!.classList.remove('d-none');
+}
```

Comment:
Previous review. 

These two functions can be merged into `toggleComment`. The methods can be called as (example):

```is
select('.minimized-comment', comment)!.classList.toggle('d-none', toggle);
select('.minimized-comment .Details-element', comment)![toggle ? 'setAttribute' : 'remove']('open', 'true');
``` 

If TS gives you grief on the second line, use a regular IF

Reply:
classList has its specific `toggle` method

Reply:
`toggle` doesn't work on initial page loads, actually we are not toggling attributes here, we set specific classes and attributes based on the value of `minimize`.

Consider that case where `toggleComment` was called for a comment with the same argument values, `toggle` fails in that case.

Reply:
`classList.toggle(name, bool)` works exactly the same as `classList[bool ? 'add' : 'remove'](name)`, so I don't know where you're seeing this difference.

`toggleAttribute` however doesn't exist so that line can stay as is.

Reply:
I just found out that [`toggleAttribute`](https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute) does exist so all the ternary operators can be dropped